### ISSUE TITLE
parsing Dirac notation of bra ket from Latex

### ISSUE THREAD

**Issue by bhpayne** [sympy/sympy#19854]
Sympy is capable of handling the bra ket Dirac notation and printing that to Latex. However, the Latex parser does not seem to handle conventional representations for bra ket notation.

    >>> import sympy
    >>> sympy.__version__
    '1.5.1'
    >>> from sympy.parsing.latex import parse_latex

First I'll try parsing a bra

    >>> sympy.srepr(parse_latex("\\langle a |"))
    "Mul(Symbol('a'), Symbol('langle'))"

That output isn't what I intended. What if I convert the bra to Latex?

    >>> sympy.latex(Bra('a'))
    '{\\left\\langle a\\right|}'

That works. Can I use that as input to the Latex parser?

    >>> sympy.srepr(parse_latex("{\\left\\langle a\\right|}"))
      ...
      File "/usr/local/lib/python3.6/dist-packages/sympy/parsing/latex/_parse_latex_antlr.py", line 57, in syntaxError
        raise LaTeXParsingError(err)
    sympy.parsing.latex.errors.LaTeXParsingError: I don't understand this
    {\left\langle a\right|}
    ~~~~~~~~~~~~~~~~~~~~~^

**Comment by oscarbenjamin** [comment]
I don't think that the parser should handle this by default but it could be an optional feature.

**Comment by iammosespaulr** [comment]
I've added support for that on my fork ... you could cherry pick 8b7b470954238cd5b7c1861d6f4cdf8cd81d4072 if you want 

---
![Screen Shot 2020-08-06 at 8 12 49 PM](https://user-images.githubusercontent.com/28682735/89546133-dc506c80-d821-11ea-8c98-3ad688acfc2e.png)

**Comment by bhpayne** [comment]
Hmm, I'm not doing something right:

    git clone https://github.com/iammosespaulr/sympy.git
    ... make Docker container that includes above sympy repo ...

    import sympy
    from sympy.parsing.latex import parse_latex
    sympy.srepr(parse_latex(r"\langle a \right|"))
"Mul(Symbol('langle'), Mul(Symbol('a'), Symbol('right')))"
    sympy.srepr(parse_latex(r"\left\langle a \right|"))
"Mul(Symbol('left'), Mul(Symbol('langle'), Mul(Symbol('a'), Symbol('right'))))"

**Comment by bhpayne** [comment]
Figured out my mistake -- I was on the wrong branch. Here's what I get now:

    >>> import sympy
    >>> sympy.__version__
    '1.7.dev'
    >>> from sympy.parsing.latex import parse_latex
    >>> sympy.srepr(parse_latex(r"\langle a \right|"))
    "Bra(Symbol('a'))"
    >>> sympy.srepr(parse_latex(r"\left\langle a \right|"))
    "Bra(Symbol('a'))"
